flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -123,9 +142,115 @@ public Builder serializer(BigQueryProtoSerializer serializer) {
             return this;
         }
 
+        public Builder streamExecutionEnvironment(
+                StreamExecutionEnvironment streamExecutionEnvironment) {
+            this.env = streamExecutionEnvironment;
+            return this;
+        }
+
         public BigQuerySinkConfig build() {
+            if (deliveryGuarantee == DeliveryGuarantee.EXACTLY_ONCE) {
+                validateStreamExecutionEnvironment(env);
+            }
             return new BigQuerySinkConfig(
                     connectOptions, deliveryGuarantee, schemaProvider, serializer);
         }
     }
+
+    // DO NOT USE!
+    // This method is used internally to create sink config for the Table API integration.
+    // Note that the serializer is hard coded for Flink Table's RowData.
+    @Internal
+    public static BigQuerySinkConfig forTable(
```

Comment:
can we make this private? i'm sure it isn't private for a reason tho

Reply:
This is public because the usage is in [Table API sink](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/pull/168/files#diff-f10948420094e406274f40c40309831f241111c27678bfb05af55049887ff8ff), where the only relation that can exist between the two classes is composition.

Moreover, a private method will still need to be internally invoked by some other exposed method. So, the idea is for this method to be internal, and useless for the connector's users. Since it is used by the Table API sink, the method's hard coding of serializer and schema derivation ensure that it can serve no other purpose.